File: paperwriter/backend/test_regex.py
import re

latex = r'''
\section{Introduction}
Welcome to PaperWriter!

\section{Methodology}
Our methodology...

\end{document}
'''
pattern = r'\\(section|subsection|subsubsection)\{([^}]+)\}([\s\S]*?)(?=\\(section|subsection|subsubsection)\{|\\bibliographystyle|\\end\{document\})'
matches = re.findall(pattern, latex)
for m in matches:
    print("Found section:", m[1])

File: paperwriter/backend/test_regex.js
const latex = 
\\section{Introduction}
Welcome to PaperWriter!

\\section{Methodology}
Our methodology...

\\bibliographystyle{IEEEtran}
\\bibliography{references}
\\end{document}
;
const sectionRegex = /\\(section|subsection|subsubsection)\{([^}]+)\}([\s\S]*?)(?=\\(section|subsection|subsubsection)\{|\\bibliographystyle|\\end\{document\})/g;
let match;
while ((match = sectionRegex.exec(latex)) !== null) {
    console.log("Found section:", match[2]);
}
